### DISCUSSION TITLE
How to export declaration files from a package

### DISCUSSION THREAD

**Discussion by vctormb** [vercel/turborepo#4578] [Help] [answered]
Hello everyone!

I have a monorepo where it contains `apps/web` and `packages/ui`. The `packages/ui` contains components from material-ui and the `apps/web` uses those components by importing them from `ui`.

In `packages/ui` I had to create a folder called `@types/@mui/material/styles.d.ts` to overwrite the theme type. The `d.ts` file works perfectly inside its own folder `packages/ui`, but when I import the theme in `apps/web` the `d.ts` types are not working.

Example:

packages/ui/index.ts:
```ts
export const theme: ThemeOptions = {
  palette: {
    white: { // <- new property added. the d.ts file has this new property
      main: colors.white,
    },
};
```

apps/web:
```ts
import { theme } from 'ui';

const color = theme.palette.white.main // ts says that `white` property doesn't exist
```

but if I add a reference it works.
packages/ui/index.ts:
```ts
/// <reference path="./@types/@mui/material/styles.d.ts" />

export const theme: ThemeOptions = { ... };
```

I tried to include the `d.ts` file in `packages/ui/tsconfig.json` but it didn't work

```json
"include": ["./@types/@mui/material/styles.d.ts", "."],
```

and the package json is like this:
```json
{
  "name": "ui",
  "version": "0.0.0",
  "main": "./index.ts",
  "types": "./index.ts",
```

Is there a better way to avoid using `/// <reference` in the index file?

**UPDATE:**
In `apps/web/tsconfig.json` I included the `d.ts` from `packages/ui`, like this and it worked (idk if this is a great solution):
```json
{
 // ...
  "include": [
    "../../packages/ui/@types/@mui/material/styles.d.ts",
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
// ...
}

```

Thanks everyone! 🙏

**Comment by anthonyshew** [comment, marked as answer]
Looks like you answered your own question! Will close.
